blueocean-dashboard/src/main/js/components/RunDetailsArtifacts.jsx

```diff
@@ -27,33 +83,41 @@ export default class RunDetailsArtifacts extends Component {
 
         const headers = [
             { label: t('rundetail.artifacts.header.name', { defaultValue: 'Name' }), className: 'name' },
-            { label: t('rundetail.artifacts.header.size', { defaultValue: 'Header' }), className: 'size' },
+            { label: t('rundetail.artifacts.header.size', { defaultValue: 'Size' }), className: 'size' },
             { label: '', className: 'actions' },
         ];
 
         const style = { fill: '#4a4a4a' };
 
         return (
-            <Table headers={headers} className="artifacts-table">
-                { artifacts.map(artifact => (
-                    <tr key={artifact.url}>
-                        <td>{artifact.name}</td>
-                        <td>
-                            <FileSize bytes={artifact.size} />
-                        </td>
-                        <td className="download">
-                            <a target="_blank" title={t('rundetail.artifacts.button.download', { defaultValue: 'Download the artifact' })} href={artifact.url}>
-                                <Icon style={style} icon="file_download" />
-                            </a>
-                        </td>
-                    </tr>
-                ))}
-            </Table>
+            <div>
+                <ArtifactListingLimited artifacts={artifacts} t={t} />
+                <Table headers={headers} className="artifacts-table">
+                    { artifacts.map(artifact => (
+                        <tr key={artifact.url}>
+                            <td>{artifact.path}</td>
+                            <td>
+                                <FileSize bytes={artifact.size} />
+                            </td>
+                            <td className="download">
+                                <a target="_blank" title={t('rundetail.artifacts.button.download', { defaultValue: 'Download the artifact' })} href={`${UrlConfig.getJenkinsRootURL()}${artifact.url}`}>
```

Comment:
Another long line :)  (sorry!)  Might be worth just building the href before actually returning the JSX.